api/src/lib/utils.ts

```diff
@@ -29,3 +33,64 @@ export function errorToJson(error: Error) {
     // Optionally add more properties here if needed
   };
 }
+
+export function shouldIgnoreFile(
```

Comment:
factored out ignore logic here, but got too lazy to write unit tests... but now i'm writing this i realize i could just write some with cursor

Reply:
okay tested!